src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,369 @@
+        {
+            return false;
+        }
+
+        if (locationInfo.positionTag != aOther.locationInfo.positionTag)
+        {
+            return false;
+        }
+
+        if (locationInfo.surfaceTag != aOther.locationInfo.surfaceTag)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @return The location name.
+     */
+    CharSpan GetName()
+    {
+        if (locationInfo.locationInfo.IsNull())
+        {
+            return { mLocationNameBuffer, 0 };
+        }
+
+        return locationInfo.locationInfo.Value().locationName;
+    }
+
+private:
+    char mLocationNameBuffer[kLocationNameMaxSize];
+};
+
+/**
+ * This class wraps the MapStruct object and provides a more user-friendly interface for the data.
+ */
+struct MapStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::MapStruct::Type
+{
+    MapStructureWrapper() { Set(0, CharSpan()); }
+
+    /**
+     * @brief This is a full constructor that initializes the map object with the given values. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note If aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    MapStructureWrapper(uint8_t aMapId, const CharSpan & aMapName) { Set(aMapId, aMapName); }
+
+    /**
+     * @brief This is a copy constructor that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper(const MapStructureWrapper & aOther) { *this = aOther; }
+
+    /**
+     * @brief This is an assignment operator that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper & operator=(const MapStructureWrapper & aOther)
+    {
+        Set(aOther.mapID, aOther.name);
+        return *this;
+    }
+
+    /**
+     * @brief Set all fields of the map object. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note if aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    void Set(uint8_t aMapId, const CharSpan & aMapName)
+    {
+        mapID = aMapId;
+
+        // Save the map name to the buffer so that access and lifetime is controlled.
+        // Create a CharSpan reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
```

Comment:
Again, this comment does not make sense.

Reply:
I don't think that this memset is needed at all. I have removed them [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/d5f855ed193df1a503fbb40f64c80f2d6fb8bfe9).